src/hook.rs

```diff
@@ -74,51 +74,85 @@ pub fn load_env(
         Some(x) => Some(Hash::from_str(x)?),
     };
 
-    let target: Option<Source> = load_trusted_source(pathbuf)?;
+    // "targets" are sources of shadowenv lisp files
+    let targets: Option<Vec<Source>> = load_trusted_sources(pathbuf)?;
 
-    match (&active, &target) {
+    let targets_hash = hash_sources(&targets);
+
+    // before we had multiple targets, this ensured we only act if we needed to
+    match (&active, &targets) {
+        // if there is no active shadowenv and we've got no targets, then we have nothing to compute
         (None, None) => {
             return Ok(None);
         }
-        (Some(a), Some(t)) if a.hash == t.hash()? && !force => {
+        // if there is an active shadowenv and some action we've taken leads us to still be in the same one, we do nothing
+        // unless the foce flag was specified
+        // probably need to update whatever sets prev_hash to be a hash of all the targets' hashes (?)
+        (Some(a), Some(t)) if a.hash == targets_hash.unwrap() && !force => {
```

Comment:
```suggestion
        (Some(a), Some(_)) if a.hash == targets_hash.unwrap() && !force => {
```
t doesn't seem to get used here